apps/dalmatiner_db/src/dalmatiner_tcp.erl

```diff
@@ -76,7 +75,20 @@ loop(Socket, Transport, State) ->
                                         dict = bkt_dict:new(Bucket,
                                                             State#state.n,
                                                             State#state.w)},
+                                {incomplete, <<>>});
+                {stream, Bucket, Delay, Resolution} ->
+                    lager:info("[tcp] Entering stream mode for bucket '~s' "
+                               "with max delay: ~p and resolution: ~pms",
+                               [Bucket, Delay, Resolution]),
+                    ok = dalmatiner_opt:set_resolution(Bucket, Resolution),
```

Comment:
No there are no errors returned in other cases but that has good reasons, all other case are purely client errors that have nothing to do with server state. A malformated package is a malformated package on the client too.

However when the client asking to set resolution X it has no way of knowing if that will work of not, if it has been set before or by someone entirely different, that's why returning an error from the server is the sensible thing to do here.